src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -228,8 +248,31 @@ def parseListV(volumes) {
     return listV
 }
 
-def runK8(image, branchName, config, axis) {
+Map getArchConf(arch) {
+    def nodeSelector = ''
+    def jnlpImage = ''
+    def archConfMap = [:]
+
+    switch(arch) {
```

Comment:
it should extract nodeSelector and other key from config.kubernetes.arch_table which should always exist (or from yaml file or as populates with defaults

Reply:
Fixed.